### PR TITLE
Help page follows theme - fixes #1121

### LINKED ISSUE #1121: Help page should respect dark theme
The Tridactyl help page should respect the setting of `theme` when set to `dark`.

### PR DESCRIPTION
**Pull request by rprimus**
Mon Oct 12 12:47:14 BST 2020

Fix for #1121 .

How should the final changes be made to `main.css`?

Currently, I've added config sections that override the current settings.
Should these changes be merged into their corresponding sections?

### CODE REVIEW

**Comment by bovine3dom** [comment]
Let's merge them with the current CSS - git will tell us what we changed if we ever needed to update.

While you're there could you make the scroll bar follow the theme? https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Scrollbars

**Comment by rprimus** [comment]
Mon Oct 12 14:22:53 BST 2020

#### greenmat-scrollbar

tridactyl-help-greenmat-scrollbar

#### quake-scrollbar

tridactyl-help-quake-scrollbar

#### default-scrollbar

tridactyl-help-default-scrollbar

**Comment by bovine3dom** [comment]
that scroll bar could be prettier IMHO.

Now I think about it, Firefox should pick one automatically that looks OK - https://bugzilla.mozilla.org/show_bug.cgi?id=1578377 - so perhaps we can somehow provide it with enough information to do that?

**Comment by rprimus** [comment]
Mon Oct 12 15:42:37 BST 2020

Which one?
The default themed one does look harsh but the others look fine to me.

Scan of the articlebug report made my head hurt.   Will have a re-read tomorow.

**Comment by bovine3dom** [comment]
I think they're all ugly, but, OTOH, I think all the non-default themes are ugly (sue me) :)

I've just realised that this change is going to upset my favourite Tridactyl user - me - as I use a "dark" theme, quake, but only so that I get the commandline at the top of the screen. I don't think I want the help pages to be dark; it's annoying enough that the new tab page is dark. So while your head is hurting from scroll bars, I might make my head hurt by trying to work out how I can put this change behind a config gate : )

**Comment by rprimus** [comment]
Mon Oct 12 16:45:23 BST 2020

Think I found a solution.  Let me know what you think.

File: src/static/themes/dark/dark.css [hunk 1]
```diff
@@ -232,6 +232,8 @@
     --tridactyl-fg: var(--tridactyl-photon-colours-in-content-page-color);
     --tridactyl-bg: var(--tridactyl-photon-colours-in-content-page-background);
 
+    --tridactyl-scrollbar-color: var(--tridactyl-bg) var(--tridactyl-fg);
```
Comment (bovine3dom) [inline src/static/themes/dark/dark.css:235]: In theory I like this solution, but it doesn't seem to do anything on my machine - the scroll bar appears to be unchanged from the default.

Comment (rprimus) [reply src/static/themes/dark/dark.css:235]: Tue Oct 13 16:48:21 BST 2020

Are you testing on the `help` page?

#### dark: bg: pink

tridactyl-help-dark-bg=pink

#### dark: bg: darkgreen

tridactyl-help-dark-bg=darkgreen

The above were set using the developer tools to change the colours.  The scrollbar changes to match.

My workflow - after making changes to `css` files:
```sh
: ; rm -rf generated build && yarn run build && yarn run run
```
##### Note:
So far, the `scrollbar` changes only affect the `help` page.

Comment (bovine3dom) [reply src/static/themes/dark/dark.css:235]: > My workflow - after making changes to css files

Aha, that was my problem, somehow only some of the CSS was getting updated. It works for me now

I'll have a final look tomorrow and then, if all goes well, merge it.

**Comment by rprimus** [comment]
On Wed, Oct 14, 2020 at 12:47:38PM -0700, Oliver Blanthorn wrote:
>
>I'll have a final look tomorrow and then, if all goes well, merge it.

Thu 15 Oct 2020 03:54:54 BST

Cool.  Can someone (with good aesthetics) choose appropriate colours for
the scrollbar in `dark` and `quake` themes?


-- 
-primus (Train yourself and be your own master.)
"First, solve the problem.  Then, write the code." - John Johnson
"Narrowness of experience leads to narrowness of imagination." - Rob Pike
http://www.catb.org/esr/faqs/smart-questions.html
http://www.chiark.greenend.org.uk/~sgtatham/bugs.html

**Comment by bovine3dom** [comment]
It doesn't change the scrollbar on the new tab page - it probably should, right?

**Comment by bovine3dom** [comment]
> Cool.  Can someone (with good aesthetics) choose appropriate colours for the scrollbar in `dark` and `quake` themes?

I've picked different colours for them at least : )

**Comment by rprimus** [comment]
Tue Oct 20 11:49:00 BST 2020

> It doesn't change the scrollbar on the new tab page - it probably should, right?

Changes to remaining pages: (main) scrollbar follows theme.

#### Questions:

1. In `shydactyl.css`, `TridactylOwnNamespace` preceeds the theme class, however, in all other `CSS` files, it is the last 
in the chain.  
a.  Is there a reason for the difference here?
b.  In general (when chaining classes), does the order make a difference?

**Comment by rprimus** [comment]
Mon Nov  2 08:55:13 GMT 2020

Hi,

What's the next step in the process for this PR?

**Comment by bovine3dom** [comment]
I thought I'd merged it ... sorry, has been a hectic couple of weeks. I'll try to merge it today.

Thanks for reminding me!

**Comment by bovine3dom** [comment]
> In shydactyl.css, TridactylOwnNamespace preceeds the theme class, however, in all other CSS files, it is the last
in the chain.

I have no idea. quake.css looks the same but I see that others are different. I'll try to work this out : )

**Comment by bovine3dom** [comment]
Merged. Thanks and sorry that I took so long!

**Pull request merged**